### ISSUE TITLE
Add more testing.

### ISSUE THREAD

**Issue by kbenzie** [8BitPimp/tack#13]
Testing is currently very sparse, we should at least have one test case per intrinsic used. I'll work on this on my flight to Seoul.

**Comment by bit-hack** [comment]
Yeah most of the tests so far have just been very basic smoke tests.  This would be great!  I'll have a think about testing too.  The tests we have so far have already been super useful for catching some of my derps :D

**Comment by kbenzie** [comment]
Added some more tests in #20 however we still don't trigger all entry points.
